File: 백준/Silver/5525. IOIOI/IOIOI.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int m = Integer.parseInt(br.readLine());
        int length = 2 * n + 1;
        String word = br.readLine();
        int cnt = 0;
        int IOcnt = 0;
        char prev = 0;

        for (int i = 0; i < m; i++) {
            char cur = word.charAt(i);

            if (IOcnt == 0 && cur == 'O') {
                continue;
            } else if (IOcnt == 0) {
                IOcnt += 1;
                prev = cur;
            }

            if (prev == cur) {
                if (IOcnt >= length) {
                    cnt += (IOcnt - length) / 2 + 1;
                }

                if (cur == 'I') {
                    IOcnt = 1;
                } else {
                    IOcnt = 0;
                    prev = 0;
                }
            } else {
                IOcnt += 1;
                prev = cur;
            }
        }

        if (IOcnt >= length) {
            cnt += (IOcnt - length) / 2 + 1;
        }

        System.out.println(cnt);
    }
}

File: 백준/Silver/5525. IOIOI/IOIOI.py
# IOIOI
import sys
input = sys.stdin.readline

n = int(input())
m = int(input())
length = 2 * n + 1
string = input().rstrip()
cnt = 0

tmp = []
for i in range(m):
    cur = string[i]
    if len(tmp) == 0 and cur == "O":
        continue
    elif len(tmp) == 0:
        tmp.append("I")
        continue
    
    if tmp[-1] == cur:
        if len(tmp) >= length:
            cnt += (len(tmp) - length) // 2 + 1 

        if cur == "I":
            tmp = ["I"]
        else:
            tmp = []
        continue
    
    else:
        tmp.append(cur)

if len(tmp) >= length:
    cnt += (len(tmp) - length) // 2 + 1
    
print(cnt)
